ouroboros-consensus/src/Ouroboros/Consensus/Node/Recovery.hs

```diff
@@ -37,11 +37,14 @@ createMarkerOnCleanShutdown ::
      (IOLike m, StandardHash blk, Typeable blk)
   => Proxy blk
   -> HasFS m h
+  -> StrictTVar m Bool
   -> m a  -- ^ Action to run
   -> m a
-createMarkerOnCleanShutdown pb mp = onExceptionIf
+createMarkerOnCleanShutdown pb mp shouldCreate = onExceptionIf
     (not . exceptionRequiresRecovery pb)
```

Comment:
So the problem that we had is that the `SIGINT` signal resulted in an exception that did not require recovery, is that correct? 

I wonder if one could translate this exception into one that requires recovery when we're in the initialization phase still and re-validating the chunk files.

Reply:
I think from the discussion that this is not needed in the end with the new mechanism